common/src/main/java/com/microsoft/hydralab/performance/parsers/WindowsBatteryResultParser.java

```diff
@@ -2,14 +2,227 @@
 // Licensed under the MIT License.
 package com.microsoft.hydralab.performance.parsers;
 
+import cn.hutool.core.collection.ConcurrentHashSet;
+import com.microsoft.hydralab.performance.PerformanceInspectionResult;
 import com.microsoft.hydralab.performance.PerformanceResultParser;
 import com.microsoft.hydralab.performance.PerformanceTestResult;
+import lombok.Data;
+import lombok.NonNull;
+import org.apache.commons.io.input.ReversedLinesFileReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * To change the default sampling interval, from an elevated command line, run: `reg add
+ * "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\SRUM\Parameters" /v Tier1Period /t REG_DWORD /d
+ * <seconds> /f`. Then restarting the PC to make it take effect.
+ *
+ * NOTE:
+ * 1. To minimize the overhead, you should gather data as infrequently as possible for your measurement. Changing the
+ * sampling rate affects all SRUM providers and should only be done for testing purposes. Changing Tier1Period to < 10
+ * seconds or > 180 seconds will result in the default value of 60 seconds being used.
+ * 2. It is recommended to start parsing the result file 20 seconds after the last power data collection. This is
+ * because even though the execution of the corresponding powershell has done, the results may not have been written to
+ * the hard disk completely because the OS has not yet completed the last step of writing due to big size of the result
+ * file is too large.
+ */
+@Data
+class WindowsBatteryParsedData {
```

Comment:
nit: I suggest we extract this class to `com.microsoft.hydralab.performance.Entity`. As we may use it for UI visualization later.

Reply:
Cool. They will be extracted to WindowsBatteryParsedData.java and WindowsMemoryParsedData.java